src/mlpack/core/tree/rectangle_tree/rectangle_tree.hpp

```diff
@@ -76,6 +76,9 @@ class RectangleTree
   } SplitHistoryStruct;
 
  private:
+  //! The max number of child nodes a non-leaf normal node can have. 
+  //! (used in x-trees)
+  size_t normalNodeMaxNumChildren;
```

Comment:
I don't like putting X tree logic in the generic RectangleTree class; I think maybe the right thing here is some refactoring.  I wonder if maybe `CondenseTree()` should call some function in `SplitType`.  But probably the refactoring that has to happen to make this work is pretty significant, so if you can't think of something now we can open a ticket and save it for some other time.  I read through `CondenseTree()` but it was only a quick read and I didn't think of any easy solutions for refactoring this x-tree logic out.  Any ideas?